app/javascript/controllers/evaluation_form_controller.js

```diff
@@ -0,0 +1,41 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="evaluation-form"
+export default class extends Controller {
+  static targets = ["challengeID", "challengePhase", "startDate"];
+
+  handleChallengeSelect(e) {
+    let id, phase, end_date
+    [id, phase, end_date] = e.target.value.split(".")
+    if (id) {
+      // set values of hidden form fields 
+      this.challengeIDTarget.value = id
+      this.challengePhaseTarget.value = phase
```

Comment:
how does this connect to the form? I'm new to Stimulus and this all seems magical to me